modules/redpanda/redpanda_test.go

```diff
@@ -701,6 +713,86 @@ func TestRedpandaBootstrapConfig(t *testing.T) {
 	}
 }
 
+func TestRedpandaHTTPProxy(t *testing.T) {
+	ctx := context.Background()
+
+	ctr, err := redpanda.Run(ctx, testImage)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+
+	httpCl := &http.Client{Timeout: 10 * time.Second}
+
+	// Get HTTP Proxy URL
+	httpProxyURL, err := ctr.HTTPProxyAddress(ctx)
+	require.NoError(t, err)
+
+	// Test getting list of topics
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, httpProxyURL+"/topics", nil)
+	require.NoError(t, err)
+	resp, err := httpCl.Do(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+
+	var topics []string
+	err = json.NewDecoder(resp.Body).Decode(&topics)
+	require.NoError(t, err)
+
+	// Test getting brokers list
+	req, err = http.NewRequestWithContext(ctx, http.MethodGet, httpProxyURL+"/brokers", nil)
+	require.NoError(t, err)
+	resp, err = httpCl.Do(req)
+	require.NoError(t, err)
+	defer resp.Body.Close()
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+
+	var brokers map[string]any
+	err = json.NewDecoder(resp.Body).Decode(&brokers)
+	require.NoError(t, err)
+	require.Contains(t, brokers, "brokers")
+}
+
+func TestRedpandaHTTPProxyWithAuthentication(t *testing.T) {
```

Comment:
I'd rename this test to be more expressive:
```suggestion
func TestHTTPProxyWithBasicAuthentication(t *testing.T) {
```

I'd also add a test for OIDC, if possible 🙏